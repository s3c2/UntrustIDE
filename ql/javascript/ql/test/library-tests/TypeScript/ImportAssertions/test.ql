import javascript

query Expr getImportAttributesFromImport(ImportDeclaration decl) {
  result = decl.getImportAttributes()
}

query Expr getImportAttributesFromExport(ExportDeclaration decl) {
  result = decl.getImportAttributes()
}

query Expr getImportOptions(DynamicImportExpr imprt) { result = imprt.getImportOptions() }

query JSParseError errors() { any() }
