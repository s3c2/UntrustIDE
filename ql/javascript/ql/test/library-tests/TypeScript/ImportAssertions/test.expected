getImportAttributesFromImport
| js-import-assertions.js:1:1:1:38 | import  ... son" }; | js-import-assertions.js:1:22:1:37 | { type: "json" } |
| js-import-assertions.js:2:1:2:51 | import  ... son" }; | js-import-assertions.js:2:35:2:50 | { type: "json" } |
| js-import-assertions.js:3:1:3:50 | import  ... son" }; | js-import-assertions.js:3:34:3:49 | { type: "json" } |
| js-import-assertions.js:4:1:4:46 | import  ... son" }; | js-import-assertions.js:4:30:4:45 | { type: "json" } |
| js-import-assertions.js:15:1:15:40 | import  ... son" }; | js-import-assertions.js:15:24:15:39 | { type: "json" } |
| js-import-assertions.js:16:1:16:53 | import  ... son" }; | js-import-assertions.js:16:37:16:52 | { type: "json" } |
| js-import-assertions.js:17:1:17:52 | import  ... son" }; | js-import-assertions.js:17:36:17:51 | { type: "json" } |
| js-import-assertions.js:18:1:18:48 | import  ... son" }; | js-import-assertions.js:18:32:18:47 | { type: "json" } |
| ts-import-assertions.ts:3:1:3:40 | import  ... son" }; | ts-import-assertions.ts:3:24:3:39 | { type: "json" } |
| ts-import-assertions.ts:4:1:4:53 | import  ... son" }; | ts-import-assertions.ts:4:37:4:52 | { type: "json" } |
| ts-import-assertions.ts:5:1:5:52 | import  ... son" }; | ts-import-assertions.ts:5:36:5:51 | { type: "json" } |
| ts-import-assertions.ts:6:1:6:48 | import  ... son" }; | ts-import-assertions.ts:6:32:6:47 | { type: "json" } |
getImportAttributesFromExport
| js-import-assertions.js:6:1:6:50 | export  ... son" }; | js-import-assertions.js:6:34:6:49 | { type: "json" } |
| js-import-assertions.js:7:1:7:45 | export  ... son" }; | js-import-assertions.js:7:29:7:44 | { type: "json" } |
| js-import-assertions.js:8:1:8:51 | export  ... son" }; | js-import-assertions.js:8:35:8:50 | { type: "json" } |
| js-import-assertions.js:20:1:20:52 | export  ... son" }; | js-import-assertions.js:20:36:20:51 | { type: "json" } |
| js-import-assertions.js:21:1:21:47 | export  ... son" }; | js-import-assertions.js:21:31:21:46 | { type: "json" } |
| js-import-assertions.js:22:1:22:53 | export  ... son" }; | js-import-assertions.js:22:37:22:52 | { type: "json" } |
| ts-import-assertions.ts:8:1:8:52 | export  ... son" }; | ts-import-assertions.ts:8:36:8:51 | { type: "json" } |
| ts-import-assertions.ts:9:1:9:47 | export  ... son" }; | ts-import-assertions.ts:9:31:9:46 | { type: "json" } |
| ts-import-assertions.ts:10:1:10:53 | export  ... son" }; | ts-import-assertions.ts:10:37:10:52 | { type: "json" } |
getImportOptions
| js-import-assertions.js:10:12:10:55 | import( ... n" } }) | js-import-assertions.js:10:29:10:54 | { with: ... on" } } |
| js-import-assertions.js:24:12:24:57 | import( ... n" } }) | js-import-assertions.js:24:29:24:56 | { asser ... on" } } |
| ts-import-assertions.ts:12:12:12:57 | import( ... n" } }) | ts-import-assertions.ts:12:29:12:56 | { asser ... on" } } |
errors
